src/lab/Task8.java

```diff
@@ -0,0 +1,364 @@
+                            """);
+                    int numberOfNewNumbers = sc.nextInt();
+                    movies = copyArray(movies, movieArraySize + numberOfNewNumbers);
+                    insertMovie(sc, movies, movieArraySize);
+                    movieArraySize += numberOfNewNumbers;
+                    break;
+                }
+                case 4: {
+                    System.out.println("""
+                            
+                            You selected to view the statistics!""");
+
+
+                    // Finding MAX using Stream
+                    /**
+                     Movie movieWithMaxDouble = Arrays.stream(movies)
+                     .max(Comparator.comparingDouble(Movie::getRating))
+                     .orElse(null);
+                     System.out.println(movieWithMaxDouble.toString());
+                     **/
+
+                    if (movies.length == 0) {
+                        System.out.println("""
+                                
+                                Unfortunately there is no movie to retrieve statistics :(""");
+                        break;
+                    } else if (movies.length == 1) {
+                        System.out.printf("""
+                                        
+                                        Movie name with the max rating: %s
+                                        Movie name with the min rating: %s
+                                        Movie rating average: %.2f%n""",
+                                movies[0].getName(),
+                                movies[0].getName(),
+                                movies[0].getRating()
+                        );
+                        break;
+                    }
+                    System.out.printf("""
+                                    
+                                    Movie name with the max rating: %s
+                                    Movie name with the min rating: %s
+                                    Movie rating average: %.2f%n""",
+                            findMaxOrMin(movies, true),
+                            findMaxOrMin(movies, false),
+                            findAverage(movies)
+                    );
+                    break;
+                }
+                case 5: {
+                    System.out.println("""
+                            
+                            You selected to search for a movie!""");
+                    if (movies.length == 0) {
+                        System.out.println("""
+                                
+                                Unfortunately there is no movie to search for :(""");
+                    } else {
+                        System.out.println("""
+                                
+                                Type the name of the movie you would like to search for:""");
+                        sc.nextLine();
+                        String movieName = sc.nextLine().trim().toLowerCase();
```

Comment:
you already use `equalsIgnoreCase `in findMovie, then no need `toLowerCase `here